plugins/entity_points_converter.py

```diff
@@ -0,0 +1,372 @@
+        ),
+        location="query",
+        required=False,
+    )
+    @CONVERTER_BLP.require_jwt("jwt", optional=True)
+    def get(self, errors):
+        """Return the micro frontend."""
+        return self.render(request.args, errors)
+
+    @CONVERTER_BLP.html_response(
+        HTTPStatus.OK, description="Micro frontend of the hello world plugin."
+    )
+    @CONVERTER_BLP.arguments(
+        InputParametersSchema(
+            partial=True, unknown=EXCLUDE, validate_errors_as_result=True
+        ),
+        location="form",
+        required=False,
+    )
+    @CONVERTER_BLP.require_jwt("jwt", optional=True)
+    def post(self, errors):
+        """Return the micro frontend with prerendered inputs."""
+        return self.render(request.form, errors)
+
+    def render(self, data: Mapping, errors: dict):
+        schema = InputParametersSchema()
+        return Response(
+            render_template(
+                "simple_template.html",
+                name=Converter.instance.name,
+                version=Converter.instance.version,
+                schema=schema,
+                values=data,
+                errors=errors,
+                process=url_for(f"{CONVERTER_BLP.name}.CalcView"),
+            )
+        )
+
+
+@CONVERTER_BLP.route("/process/")
+class CalcView(MethodView):
+    """Start a long running processing task."""
+
+    @CONVERTER_BLP.arguments(InputParametersSchema(unknown=EXCLUDE), location="form")
+    @CONVERTER_BLP.response(HTTPStatus.OK, TaskResponseSchema())
+    @CONVERTER_BLP.require_jwt("jwt", optional=True)
+    def post(self, arguments):
+        """Start the calculation task."""
+        db_task = ProcessingTask(
+            task_name=calculation_task.name,
+            parameters=InputParametersSchema().dumps(arguments),
+        )
+        db_task.save(commit=True)
+
+        # all tasks need to know about db id to load the db entry
+        task: chain = calculation_task.s(db_id=db_task.id) | save_task_result.s(
+            db_id=db_task.id
+        )
+        # save errors to db
+        task.link_error(save_task_error.s(db_id=db_task.id))
+        task.apply_async()
+
+        db_task.save(commit=True)
+
+        return redirect(
+            url_for("tasks-api.TaskView", task_id=str(db_task.id)), HTTPStatus.SEE_OTHER
+        )
+
+
+class Converter(QHAnaPluginBase):
+
+    name = _plugin_name
+    version = __version__
+    description = "Converts a file containing entity points into another file format, e.g. csv into json"
+    tags = ["converter"]
+
+    def __init__(self, app: Optional[Flask]) -> None:
+        super().__init__(app)
+
+    def get_api_blueprint(self):
+        return CONVERTER_BLP
+
+
+TASK_LOGGER = get_task_logger(__name__)
+
+
+def json_to_csv(json_gen) -> Iterator[dict]:
+    for ent in json_gen:
+        ent_dict = {"ID": ent["ID"], "href": ent["href"]}
+        ent_dict.update({f"dim{d}": ent["point"][d] for d in range(len(ent["point"]))})

```

Comment:
Yes, that was hard to get by just reading the code. However the same entity should look the same when serialized as json or as CSV. This conversion plugin converts between different entity formats (e.g. point as a list of numbers to point as a a number of entity attributes dim*) and different serialization formats (json and csv) in one go. This should be documented (maybe in the plugin description?). The ideal solution would be to stick to entity type only so that the PCA plugin can read in both json and csv serialized entities. The "better" format in the Context of QHAna has no/less arrays. I am currently planning to formalize the entity data types (there is a draft PR for that) and <https://github.com/UST-QuAntiL/qhana-plugin-runner/blob/feature/data-type-doc/docs/data-formats/examples/entities.rst#entityvector> would be ideal for this use case. (just make sure, that attributes can be sorted lexicographically, i.e. that dim* is padded with 0 for smaller dims `dim001` as needed)

Reply:
If I may recap, the idea is to use entity/vector as data_type. Then our entities are always in the same format no matter if they are contained in a json file or a csv file.
The format of an entity being:
 
"ID": <id>, "href": <href>, "dim00": <float>, "dim01": <float>, ..., "dim12": <float>

the zero padding of dim* of course depends on the number of dimensions.

If we use this data_type, the entities would look the same, no matter if we load them from a json or a csv file. Therefore, we can remove the conversion plugin and just adapt the other plugins to also allow csv files as an input.

Reply:
That would be ideal. It consolidates code and removes the need for a whole plugin (which we would need to maintain).

Reply:
I think this is a great idea and I will get to it.

Reply:
You could use f strings with nested format specifiers to achieve the padding `f"{number:0{padding_width}}"`